actix-identity/src/identity_errors.rs

```diff
@@ -0,0 +1,69 @@
+use actix_session::{SessionGetError, SessionInsertError};
+use actix_web::{cookie::time::error::ComponentRange, http::StatusCode, ResponseError};
+
+/// This error can occur during login attempts.
+#[derive(Debug)]
+pub struct LoginError(SessionInsertError);
+
+impl std::fmt::Display for LoginError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl std::error::Error for LoginError {}
+
+impl ResponseError for LoginError {
+    fn status_code(&self) -> StatusCode {
+        StatusCode::UNAUTHORIZED
+    }
+}
+
+impl From<SessionInsertError> for LoginError {
+    fn from(error: SessionInsertError) -> Self {
+        Self(error)
+    }
+}
+
+/// This error describes all of the potential failures which can happen
+/// while retrieving an identity.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum IdentityError {
+    /// This occurs whenever no identity is found in a session.
+    MissingIdentityError(String),
```

Comment:
This doesn't need to hold a `String` if we use a different error type for the `id` method.
We can simply provide the correct error message in the `Display` implementation.